src/engine/Filter.cpp

```diff
@@ -499,6 +500,41 @@ void Filter::computeResultFixedValue(
         rhs_string = ad_utility::convertValueLiteralToIndexWord(rhs_string);
       } else if (ad_utility::isNumeric(_rhs)) {
         rhs_string = ad_utility::convertNumericToIndexWord(rhs_string);
+      } else {
+        if (getIndex().getVocab().getCaseInsensitiveOrdering()) {
+          // We have to move to the correct end of the
+          // "same letters but different case" - range
+          // to make the filters work
+          // TODO<kalmbach, schnelle>: thoroughly test this
+          // (End-To-End or unit tests? probably both but the unit tests
+          // would also be in an end-to-end fashion for those nested
+          // mechanisms).
+          switch (_type) {
+            case SparqlFilter::GE:
+            case SparqlFilter::LT: {
+              std::transform(rhs_string.begin(), rhs_string.end(),
```

Comment:
We have `ad_utility::getUppercaseUtf8()` which handles UTF8 (though it's quite ugly at the moment)